src/components/Identity/createComponent.js

```diff
@@ -0,0 +1,179 @@
+import createCustomerIds from "./customerIds/createCustomerIds";
+import { defer } from "../../utils";
+import { EXPERIENCE_CLOUD_ID } from "./constants/cookieNames";
+import migration from "./migration";
+import createEvent from "../DataCollector/createEvent";
+
+const addIdsContext = (payload, ecid) => {
+  payload.addIdentity(EXPERIENCE_CLOUD_ID, {
+    id: ecid
+  });
+};
+
+export default (idSyncs, manualIdSyncs, cookieJar, config, logger, network) => {
+  let optIn;
+  let deferredForEcid;
+  let lifecycle;
+  let customerIds;
+  let alreadyQueriedForIdSyncs = false;
+  const { migrateIds, imsOrgId } = config;
+
+  const createPayload = event => {
+    const payload = network.createPayload();
+    payload.addEvent(event);
+    return payload;
+  };
+
+  const readEcid = () => {
+    let ecid = cookieJar.get(EXPERIENCE_CLOUD_ID);
+    if (!ecid && migrateIds) {
+      ecid = migration(imsOrgId).readEcidFromAmcvCookie();
```

Comment:
Should the legacy ID be removed at this point?

Reply:
I think we shouldn't. There could be pages from the same company using legacy libraries.